connectivity/service/src/main/java/org/eclipse/ditto/connectivity/service/messaging/httppush/DefaultHttpPushFactory.java

```diff
@@ -243,10 +243,15 @@ private static ConnectionPoolSettings disambiguateByConnectionId(final ActorSyst
                         .withParserSettings(parserSettings.withHeaderValueCacheLimits(disambiguator)));
     }
 
-    private static int parseParallelism(final Map<String, String> specificConfig) {
-        return Optional.ofNullable(specificConfig.get(HttpPushFactory.PARALLELISM))
+    static int parseParallelism(final Map<String, String> specificConfig) {
+        final int parsedParallelism = Optional.ofNullable(specificConfig.get(HttpPushFactory.PARALLELISM_JSON_KEY))
                 .map(Integer::valueOf)
                 .orElse(1);
+        return determineNextPowerOfTwo(parsedParallelism);
```

Comment:
We could write this as a mapping stage `.map(this::determineNextPowerOfTwo)`. What do you think?

Reply:
Good idea, sure 👍